modules/data.atmosphere/R/debias.met.R

```diff
@@ -0,0 +1,107 @@
+##' debias.met takes source_met, train_met and output new_met that debiases source_met against train_met
+##' @name debias.met
+##' @title debias.met
+##' @export
+##' @param outfolder
+##' @param source_met - the met data that you want to be altered 
+##' @param train_met - the met data that we use to train the other dataset
+##' @param de_method - select which debias method you would like to use, options are 'mean, median 
+##' @param site.id
+##' @author James Simkins
+debias.met <- function(outfolder, source_met, train_met, site_id, de_method='mean', overwrite=FALSE, verbose=FALSE, ...){  
+  require(PEcAn.utils)
+  require(lubridate)
+  require(ncdf4)
+  outfolder = paste0(outfolder,"_site_",paste0(site_id %/% 1000000000, "-", site_id %% 1000000000))
+  
+  var = data.frame(DAP.name = c("tas","rlds","ps","rsds","uas","vas","huss","pr"),
+                   CF.name = c("air_temperature","surface_downwelling_longwave_flux_in_air","air_pressure","surface_downwelling_shortwave_flux_in_air","eastward_wind","northward_wind","specific_humidity","precipitation_flux"),
+                   units = c('Kelvin',"W/m2","Pascal","W/m2","m/s","m/s","g/g","kg/m2/s")
+  )
+  
+  #Need to pull the year from the string filename, this assumes that the year is always placed right before '.nc'
+  substrRight <- function(x, n){
+    substr(x, nchar(x)-n+1, nchar(x))
+  }
+  sub_str= substrRight(source_met, 7)
+  year = substr(sub_str,1, 4)
+  #Read in the two datasets, with the dimensions of the source dataset being named sou.list
+  sou = list()
+  sou.list = list()
+  tem = nc_open(source_met)
+  dim = tem$dim
+  for (j in 1:length(var$CF.name)){
+    sou[[j]] = ncvar_get(tem,as.character(var$CF.name[j]))
+    sou.list[[j]] = ncvar_def(name=as.character(var$CF.name[j]), units=as.character(var$units[j]), dim=dim, missval=-999, verbose=verbose)
+  }
+  lat_sou = as.numeric(ncvar_get(tem,"latitude"))
+  lon_sou = as.numeric(ncvar_get(tem,"longitude"))
+  year = as.numeric(year)
+  nc_close(tem)
+  
+#Need to try and find the closest ameriflux site and year based on the lat/lon/year given 
+#download function (lat_sou,lon_sou,year)
+#   find closest fluxnet site and then download year, if not available, try next year up from 2006 to 2015? 
```

Comment:
Thanks! I'll give the user the option to select their own, but if the argument isn't given, we can use the find the nearest neighbor function